src/providers/app-config/app-config.ts

```diff
@@ -30,11 +36,27 @@ export class AppConfigProvider {
 
   private getRemoteData = (): Promise<any> =>
     new Promise((resolve, reject) => {
-      this.httpClient.get<any>(this.environmentFile.configUrl)
+      if (this.networkState.getNetworkState() === ConnectionStatus.ONLINE) {
+        this.httpClient.get<any>(this.environmentFile.configUrl)
         .subscribe(
-          data => resolve(data),
+          (data) => {
+            this.dataStore.setItem('CONFIG', JSON.stringify(data));
+            resolve(data);
+          },
           error => reject(error),
         );
+      } else {
+        this.getOfflineConfig()
+        .then(data => resolve(data))
+        .catch(error => reject(error));
+      }
+    })
+
+  private getOfflineConfig = (): Promise<any> =>
+    new Promise((resolve, reject) => {
+      this.dataStore.getItem('CONFIG')
```

Comment:
Looks like this is already a promise so doesn't need to be wrapped in one too..

Reply:
changed